src/RouterSelector.native.js

```diff
@@ -41,28 +46,24 @@ const handleLinks = async () => {
   log.debug('handleLinks:', { params })
   try {
     const { magiclink } = params
-    if (magiclink) {
```

Comment:
why was this if removed

Reply:
put the usesavettopath back in handle links and extract handlelinks into a single native/web helper
is there any difference between the native/web handlelinks code?

Reply:
2 major differences:
1. the way the path is taken:
web:
`let path = window.location.pathname.slice(1)`
native:
`let path = DeepLinking.pathname.slice(1)`

2. the if statement which @sirpy already commented that it can be identical (removing the `path.indexOf('Marketplace') >= 0` part)
web:
`if (params && Object.keys(params).length > 0)`
native:
`if ((params && Object.keys(params).length > 0) || path.indexOf('Marketplace') >= 0)`

Reply:
1. can be switched to DeepLinking.pathname.slice(1) for both
2. we can remove of keep markeetplace for both
3. the removal of if(magiclink) is incorrect for all i can see

Reply:
refactored handleLinks into a util